sensor/kubernetes/eventpipeline/pipeline_test.go

```diff
@@ -38,28 +45,130 @@ func (s *eventPipelineSuite) TearDownTest() {
 	s.T().Cleanup(s.mockCtrl.Finish)
 }
 
-type mockListener struct{}
-
-func (m *mockListener) Start() error { return nil }
-func (m *mockListener) Stop(_ error) {}
-
 func (s *eventPipelineSuite) SetupTest() {
 	s.mockCtrl = gomock.NewController(s.T())
 
 	s.resolver = mockComponent.NewMockResolver(s.mockCtrl)
 	s.detector = mockDetector.NewMockDetector(s.mockCtrl)
 	s.reprocessor = mockReprocessor.NewMockHandler(s.mockCtrl)
+	s.listener = mockComponent.NewMockContextListener(s.mockCtrl)
+	s.outputQueue = mockComponent.NewMockOutputQueue(s.mockCtrl)
+
+	offlineMode := atomic.Bool{}
+	offlineMode.Store(true)
+
 	s.pipeline = &eventPipeline{
 		eventsC:     make(chan *message.ExpiringMessage),
 		stopSig:     concurrency.NewSignal(),
-		output:      mockComponent.NewMockOutputQueue(s.mockCtrl),
+		output:      s.outputQueue,
 		resolver:    s.resolver,
 		detector:    s.detector,
 		reprocessor: s.reprocessor,
-		listener:    &mockListener{},
+		listener:    s.listener,
+		offlineMode: &offlineMode,
 	}
 }
 
+func (s *eventPipelineSuite) write() {
+	s.outputC <- message.NewExpiring(s.pipeline.context, nil)
+}
+
+func (s *eventPipelineSuite) online() {
+	s.pipeline.Notify(common.SensorComponentEventCentralReachable)
+}
+
+func (s *eventPipelineSuite) offline() {
+	s.pipeline.Notify(common.SensorComponentEventOfflineMode)
+}
+
+func (s *eventPipelineSuite) readSuccess() {
+	msg, more := <-s.pipeline.ResponsesC()
+	s.Assert().True(more, "channel should be open")
+	s.Assert().False(msg.IsExpired(), "message should not be expired")
+}
+
+func (s *eventPipelineSuite) readExpired() {
+	msg, more := <-s.pipeline.ResponsesC()
+	s.Assert().True(more, "channel should be open")
+	s.Assert().True(msg.IsExpired(), "message should be expired")
+}
+
+func (s *eventPipelineSuite) Test_OfflineModeCases() {
+	s.T().Setenv("ROX_RESYNC_DISABLED", "true")
+
+	outputC := make(chan *message.ExpiringMessage, 10)
+	s.outputQueue.EXPECT().ResponsesC().
+		AnyTimes().Return(outputC)
+	s.outputC = outputC
+
+	s.outputQueue.EXPECT().Start().Times(1)
+	s.resolver.EXPECT().Start().Times(1)
+	s.listener.EXPECT().StartWithContext(gomock.Any()).AnyTimes()
+	s.listener.EXPECT().Stop(gomock.Any()).AnyTimes()
+
+	s.Require().NoError(s.pipeline.Start())
+	s.pipeline.Notify(common.SensorComponentEventCentralReachable)
+
+	testCases := [][]func(){
```

Comment:
[nit] Maybe we could have something here like:
```golang
testCases := []struct{
    testName string
    orderedFunctions []func()
}
```
That way we could a more meaningful message in line 130.

```golang
s.Run(testCase.testName, func() {
    for _, fn := range testCase.orderedFunctions {
        fn()
    }
})
```

Reply:
Done